### ISSUE TITLE
Misleading exception handling in StepSynchronizer.RunThread.step(...)

### ISSUE THREAD

**Issue by rubienr** [avrora-framework/avrora#185]
In

```
edu.ucla.cs.compilers.avrora.avrora.sim.clock.StepSynchronizer.RunThread.step(...)
```
1. if a timeout exception is thrown a re-throw of 
    throw Util.unimplemented();
   without further information is misleading.
2. regarding exception handling (see code fragment below) the order (even if there is no impl. but just stubs) should be:
   1. do some cleanup
   2. report exit or exit

.

```
try {
    cycles[cntr] = sim.step();
} catch (Throwable t)
{
    reportExit(sim, t);
    removeSimulator(threads[cntr]);
}
```

**Comment by rubienr** [comment]
fixed in f88951b
never the less, the simulation should exit normally on timeout exception

**Comment by rubienr** [comment]
closed with 8da32fe
